src/Application.php

```diff
@@ -149,19 +149,30 @@ public function bootstrap(array $listeners = [])
         $this->event = $event  = new MvcEvent();
         $event->setTarget($this);
         $event->setApplication($this)
-              ->setRequest($this->request)
-              ->setResponse($this->response)
+              ->setResponse($this->getResponse())
               ->setRouter($serviceManager->get('Router'));
 
+        try {
+            $event->setRequest($this->getRequest());
+        } catch (Exception $e) {
+            $event->setRequest(new Request());
+            $event->setError(self::ERROR_EXCEPTION);
+            $event->setParam('exception', $e);
+        }
+
         // Trigger bootstrap events
-        $events->trigger(MvcEvent::EVENT_BOOTSTRAP, $event);
+        if ($event->isError()) {
+            $events->trigger(MvcEvent::EVENT_DISPATCH_ERROR, $event);
+            $this->completeRequest($event);
```

Comment:
I think this would require a new event; since we're not done with bootstrap, it's entirely possible that the application is not in a fully usable state, and triggering a dispatch.error event could lead to cascading errors.